### ISSUE TITLE
Getting Authentication Failed Exception

### ISSUE THREAD

**Issue by wat5** [digerata/grails-postmark#2]
Not sure what is going on.  I have installed the plugin and configured as directed.   When I attempt to send a test email I get an Authentication Failed Exception.  Is there something else that needs to be configured?  Do I need to remove any of the grails.mail config properties?

Thanks

**Comment by digerata** [comment]
No, you don't need to remove any grails.mail config.

Just to confirm, your configuration in Config.groovy looks like:

```
postmark {
    apikey = "HexString" 
}
```

**Comment by digerata** [comment]
Sorry, I take that back.  You would see a message in your log like "Postmark API Key is not set.  See: postmark.apikey" If it wasn't set.

I'd say triple check your API key is exactly the same as in your postmark server dashboard.

**Comment by wat5** [comment]
I checked and it is the same.   I must be missing something.  Here is the exception:

javax.mail.AuthenticationFailedException
    at javax.mail.Service.connect(Service.java:319)
    at org.grails.mail.MailService.sendMail(MailService.groovy:63)
    at org.grails.mail.MailService.sendMail(MailService.groovy:46)
    at org.grails.mail.MailService$sendMail.call(Unknown Source)
    at tobis.core.MessageService.sendTest(MessageService.groovy:35)
    at tobis.core.MessageService$$FastClassByCGLIB$$47949640.invoke(<generated>)
    at net.sf.cglib.proxy.MethodProxy.invoke(MethodProxy.java:149)
    at tobis.core.MessageService$$EnhancerByCGLIB$$832e0efe.sendTest(<generated>)
    at tobis.core.MessageService$sendTest.call(Unknown Source)
    at UserController$_closure10.doCall(UserController.groovy:287)
    at UserController$_closure10.doCall(UserController.groovy)
    at org.apache.shiro.web.servlet.ShiroFilter.executeChain(ShiroFilter.java:687)
    at org.apache.shiro.web.servlet.ShiroFilter.doFilterInternal(ShiroFilter.java:616)
    at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:81)
    at java.lang.Thread.run(Thread.java:637)

My config.groovy looks like the following:
...

grails {

```
mail {
    host = "mail.authsmtp.com"
    port = 2525             
    username = "####"
    password = "######"
    props = ["mail.smtp.auth":"true"]
} 
```

}

postmark {
    apikey = "hex key"
}

...

Thanks for your help.

**Comment by digerata** [comment]
The exception: javax.mail.AuthenticationFailedException is not thrown from Postmark but from the normal SMTP library.  It looks like that is still being used instead of Postmark to send email.

What does the code look like in MessageService that actually calls sendMail?

Is this using run-app or an integration test?

**Comment by digerata** [comment]
Also, if you are in a time crunch, you can uninstall the mail plugin and it should work right away.

**Comment by wat5** [comment]
My method looks like this:

```
def sendTest( ) {
    log.debug("Message Service - sendTest ")
    Map mail = [message: "Test Message", from:'info@xyz.com', to: "ewatson@gmail.com", subject:"Test"]
    if (mail.to)
    {
        mailService.sendMail { to mail.to 
            from mail.from 
            subject mail.subject 
            body mail.message 
        }
    }

}
```

This is using run-app.

Thanks

**Comment by digerata** [comment]
Gotcha, here is the problem.  Postmark is overriding the sendMail method added to all Controllers and Services by the mail plugin.  In this case, you are calling the mail plugin's mailService directly.  

Remove the "mailService." from mailService.sendMail and you should be fine.

**Comment by wat5** [comment]
That fixed it, thanks a bunch.   I am now getting a the following error:

 [Message:Sender signature not found for email., ErrorCode:400]
2010-05-03 13:40:08,051 [http-8080-2] ERROR errors.GrailsExceptionResolver  - Remote Postmark server error: Unprocessable Entity
java.lang.Exception: Remote Postmark server error: Unprocessable Entity

I have confirmed and verified all of the stuff for the Sender Signature.   Any thoughts?

Thanks

**Comment by digerata** [comment]
Is the from address specified in your code the same as what your sender signature is setup as?

This looks like a JSON encoding error.  But I tried the exact data you have above and it worked for me.

Any chance you can fire up a packet sniffer and see what is sent to api.postmarkapp.com?  If not, you could put a log statement on line 57 of PostmarkService.groovy to print out the postBody as a string.

**Comment by wat5** [comment]
I think I found the issue :)  This is what the trace statement says:

2010-05-03 14:21:52,105 [http-8080-2] ERROR postmark.PostmarkService  - Postmark{"From":"bounceoff@flowz.com","To":"casey.watson@gmail.com","Subject":"Test","Tag":null,"HtmlBody":null,"TextBody":"Test Message"}

Thanks

**Comment by digerata** [comment]
I'll put out a fix for it shortly.

**Comment by digerata** [comment]
OH! Hah!  I was looking at the nulls in the JSON!  I didn't even notice the from hard coded!  Sheesh!  How ridiculous.

**Comment by digerata** [comment]
The fix is out in version 0.2.

```
grails install-plugin postmark 0.2
```

**Comment by digerata** [comment]
Closed by 2102161c458262241650db332a4e1f6f1fea4fac - From field not used.
